### PR TITLE
Fixes #6874: Integers in settings can be up to 8 digits to avoid bigint out of range error

### PR DESCRIPTION
**Pull request by orrabin**

### CODE REVIEW

**Comment by orrabin** [comment]
[test]

**Comment by domcleal** [comment]
Missing "#" in commit message will prevent redmine linkage.

**Comment by GregSutcliffe** [comment]
@orrabin tested, works great. Could you add a quick test for it?

**Comment by orrabin** [comment]
@GregSutcliffe I added a test for `entries_per_page` as one of the integers

File: test/unit/setting_test.rb [hunk 1]
```diff
@@ -347,6 +351,17 @@ def check_zero_value_not_allowed_for setting_name
     assert !setting.invalid?
   end
 
+  def check_length_must_be_under_8 setting_name
+    setting = Setting.find_or_create_by_name(setting_name, :default => 30)
+    setting.value = 123456789
+
+    assert setting.invalid?
+    assert setting.errors[:value].join(";").include?("is too long (maximum is 8 characters)")
```
Comment (domcleal) [inline test/unit/setting_test.rb]: `refute_valid setting, :value, "is too long (maximum is 8 characters)"` should do the same as these lines, ditto below (`assert_valid setting`)

File: test/unit/setting_test.rb [hunk 2]
```diff
@@ -340,20 +344,30 @@ def check_zero_value_not_allowed_for setting_name
...
-    assert !setting.invalid?
+    assert_valid setting
+  end
+
+  def check_length_must_be_under_8 setting_name
+    setting = Setting.find_or_create_by_name(setting_name, :default => 30)
+    setting.value = 123456789
+
+    refute_valid setting, :value
+    assert setting.errors[:value].join(";").include?("is too long (maximum is 8 characters)")
```
Comment (orrabin) [inline test/unit/setting_test.rb]: @domcleal I couldn't use `refute_valid setting, :value, "is too long (maximum is 8 characters)"` as you suggested, the test failed with `value error matching is too long (maximum is 8 characters) not found: ["is too long (maximum is 8 characters)"]` 
I changed the test above mine that was also using the old syntax and it works fine so I think it has something to do with the parenthesis in the error message

Comment (domcleal) [inline test/unit/setting_test.rb]: ah yes, since it uses .match then you'd probably do `/is too long \(maximum is 8 characters\)/` instead

**Comment by orrabin** [comment]
@domcleal I fixed the test, thanks:)

**Comment by orrabin** [comment]
[test]

**Comment by ohadlevy** [comment]
merged as c737b10 thanks @orrabin

**Pull request closed without merging**